### PR TITLE
A few more unit tests

### PR DESCRIPTION
**Pull request by lightglitch**

### CODE REVIEW

File: pf4j/pom.xml [hunk 1]
```diff
@@ -43,6 +43,12 @@
             <version>${junit.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.mockito</groupId>
```
Comment (decebals) [inline pf4j/pom.xml]: :+1: I like mockito.

File: pf4j/src/main/java/ro/fortsoft/pf4j/PluginWrapper.java [hunk 2]
```diff
@@ -54,7 +54,7 @@ public String getPluginPath() {
 	 * for this plug-in. The class loader can be used to directly access
 	 * plug-in resources and classes.
 	 */
-    public PluginClassLoader getPluginClassLoader() {
+    public ClassLoader getPluginClassLoader() {
```
Comment (decebals) [inline pf4j/src/main/java/ro/fortsoft/pf4j/PluginWrapper.java]: I know that this change makes `PluginWrapper` testable but I think that it's a drawback from the API point of view. If it's not another option and if this change doesn't break the current code (at compilation) I am ok with this modification.

File: pf4j/src/test/java/ro/fortsoft/pf4j/DefaultExtensionFactoryTest.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+import static org.junit.Assert.*;
+
+/**
+ *
+ * @author Mario Franco
+ */
+public class DefaultExtensionFactoryTest {
+
+    public DefaultExtensionFactoryTest() {
```
Comment (decebals) [inline pf4j/src/test/java/ro/fortsoft/pf4j/DefaultExtensionFactoryTest.java]: I think that we can remove empty constructor and empty `setUp`, `tearDown` methods to strip the code. Is it ok for you?

File: pf4j/src/main/java/ro/fortsoft/pf4j/PluginWrapper.java (hunk 2 above)
Comment (lightglitch) [inline pf4j/src/main/java/ro/fortsoft/pf4j/PluginWrapper.java]: It doesn't break the current code (you can check the travis build of the pull request), another option is to also mock the PluginClassLoader in the test.

File: pf4j/src/test/java/ro/fortsoft/pf4j/DefaultExtensionFactoryTest.java (hunk 3 above)
Comment (lightglitch) [inline pf4j/src/test/java/ro/fortsoft/pf4j/DefaultExtensionFactoryTest.java]: Sure.

File: pf4j/src/main/java/ro/fortsoft/pf4j/PluginWrapper.java (hunk 2 above)
Comment (lightglitch) [inline pf4j/src/main/java/ro/fortsoft/pf4j/PluginWrapper.java]: Another thing the PluginClassLoader doesn't expose any extra API only the ClassLoader public methods so I don't see a big drawback here.

Comment (decebals) [inline pf4j/src/main/java/ro/fortsoft/pf4j/PluginWrapper.java]: I am ok with this modification.

**Comment by lightglitch** [comment]
@decebals Clean up is done.

**Comment by decebals** [comment]
:+1: Thanks!

**Pull request merged**